Trello-962 Authentication Enabled Registry

bmcelvee opened the pull request:
Related PRs: 
https://github.com/openshift/openshift-docs/pull/12065 (switch signature url for TBR - merged)
https://github.com/openshift/openshift-docs/pull/12086 ([WIP] Update references to registry.access.redhat.com to registry.redhat.io)
https://github.com/openshift/openshift-docs/pull/12199 (Add TRB Variables to Configuring Your Inventory File Topic)

bmcelvee commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+
+. Log in to the new registry with a `docker login`.
++
+* Log in with user credentials:
```
In the KB article, this step is written: 
----
# docker login https://registry.redhat.io
Username: myrhusername
Password:  xxxxxxxxxxx
----

Should I change the command here to match? Same for tokens.

bmcelvee commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+
+You can use `docker login` with your credentials, username and password or authentication
+token, to access content on the new registry.
+
```
I pulled the following content from the group D demo. I changed referenced to `oc cluster up` to "build" as a place holder. How should that process occur?

Are additional steps required?

bmcelvee commented:
@bparees, this is the PR I mentioned yesterday. Would you mind taking a look, please. :) The preview build is out of date, but I can provide a new one if you'd like.

It's a super rough draft right now, and the flow might not be quite logical. For sources, I pulled content from your demo, FAQs, emails, and the KB article.

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+new registry requires authentication for access, there is a new secret in the
+{product-title} namespace called `imagestreamsecret`. When you run your build,
+your `imagestreamsecret` gets populated with your `kubernetes.io/dockerconfigjson`
+credentials.
```
just delete this last sentence.

For a "normal" install, imagestreamsecret gets populated based on ansible vars set in the inventory.

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+
+All image streams are updated to point to the new registry. Because the
+new registry requires authentication for access, there is a new secret in the
+{product-title} namespace called `imagestreamsecret`. When you run your build,
```
this needs to be literally `openshift`, not product title.

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+
+There are two places your credentials must exist:
+
+* *{product-title} namespace*. Your credentials must exist in the {product-title}
```
literally openshift is the name of the namespace.

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+* *{product-title} namespace*. Your credentials must exist in the {product-title}
+namespace so that the image streams in the {product-title} namespace can import.
+* *Your host*. Your credentials must exist on your host because Kubernetes will
+use the credentials from your host when it goes to pull images.
```
both of these things should be setup for you by the ansible installer.  it's worth explaining they need to be there, but it should also be clear that the ansible install will set it up for you (assuming you populate all the correct ansible vars.  Since those vars are in flux right now you're going to need feedback from @sdodson and @michaelgugino to make sure the exact var names are correct)

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+As long as you have the following items in place, you should be able to
+access the new registry.
+
+* You should see image import secret, `imagestreamsecret`,  in your {product-title}
```
openshift namespace

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+There are two places your credentials must exist:
+
+* *{product-title} namespace*. Your credentials must exist in the {product-title}
+namespace so that the image streams in the {product-title} namespace can import.
```
openshift namespace

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+* You should see image import secret, `imagestreamsecret`,  in your {product-title}
+namespace. That secret should have credentials that will allow you to access
+the new registry.
+* All of your cluster nodes should have a `dockerconfigjson` that allow you to
```
`.docker/config.json`
s/allow/allows/

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+
+
+[[create-authentication-tokens]]
+=== Creating Authentication Tokens
```
let's plan to have QE validate it.  The registry is GAing in a week and they should be able to test the flow then.

bmcelvee replied:
Once all of the initial reviews are complete, I'll tag QE.

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+. Copy the token.
+
+[[authenticate-against-idp]]
+=== Authenticating Against IDP
```
you can delete this whole section (IDP)

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+----
+
+[[docker-login]]
+=== Using Docker Login to Access the New Registry
```
you can delete this whole section (docker login)

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+----
+
+[[setting-registry-as-default]]
+=== Setting the New Registry as default
```
i think you can delete this whole section too.  it's not really openshift specific nor is it necessary to make things work.

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+. Navigate back to your Service Accounts.
+. Click the Service Account you created.
+. Copy the username, including the prepended string.
+. Copy the token.
```
this section needs to somehow tie into the installation instructions.  Basically you'll need to go get this username+token and then paste it into the appropriate ansible vars in your inventory file.

Again @sdodson and @michaelgugino can provide specifics.

bparees commented on dev_guide/managing_images.adoc:
```diff
@@ -586,6 +586,235 @@ instead of listing all image streams.
+
+[NOTE]
+====
+RHEL 8 will only be available on the new registry.
```
i don't know if we should be mentioning this or not.  probably need to check w/ ben breard.  (can't find his github right now, but he's bbread on #aos)

bparees commented:
this isn't really a dev topic.  it's cluster admin/configuration and/or installation.

bmcelvee commented:
Thanks, @bparees! I'll start updating right away.

bmcelvee commented on architecture/infrastructure_components/image_registry.adoc:
```diff
@@ -54,3 +54,96 @@ repositories using credentials supplied by the user. This allows {product-title}
+. Navigate to `registry.redhat.io`.
+. Log in with your RHN username and password.
+. Accept terms when prompted.
+.. If you are not immediately prompted to accept terms, you will be prompted
```
If you have already accepted "terms", then you will not be prompted/need to accept them again.

bmcelvee commented:
Update: I split the concept content and put it in the architecture topics so the same source material will be displayed in the Architecture Guide ->  Infrastructure Components -> Container Registry and Configuring Clusters -> Setting up the Registry -> Registry Overview. I added the procedural content to the Configuring Clusters guide.

bmcelvee commented:
@sdodson @sferich888 PTAL

bmcelvee commented:
Do we need to go ahead and change all references to registry.redhat.com to registry.redhat.io? Or should we wait until the existing registry is deprecated?

Are we changing this RHEL and RHEL Atomic in the etc/containers/registries.conf? Right now registries points to registry.redhat.com. 

(Cc @chrisnegus)

bparees commented:
> Do we need to go ahead and change all references to registry.redhat.com to registry.redhat.io? Or should we wait until the existing registry is deprecated?

we should probably switch over now.

> Are we changing this RHEL and RHEL Atomic in the etc/containers/registries.conf? Right now registries points to registry.redhat.com.

that i don't know... not sure who owns that.

vikram-redhat commented:
> Do we need to go ahead and change all references to registry.redhat.com to registry.redhat.io? Or should we wait until the existing registry is deprecated?

@reestr and @sferich888 - can you weigh in on this?

bmcelvee commented:
Related PR: https://github.com/openshift/openshift-docs/pull/12065

bmcelvee commented on install_config/registry/index.adoc:
```diff
@@ -45,3 +45,112 @@ If you need an enterprise-quality container image registry, Red Hat Quay is avai
+. Copy the username, including the prepended string.
+. Copy the token.
+
+[[using-service-accounts]]
```
Added steps that can be performed from the Service Accounts page.

bmcelvee commented on install_config/registry/index.adoc:
```diff
@@ -45,3 +45,112 @@ If you need an enterprise-quality container image registry, Red Hat Quay is avai
+  -p=<token>
+----
++
+Once you successfully log in, copy `~/.docker/config.json` to
```
@sdodson this is the step that you recommended on the KB article. Is it still appropriate here?

bmcelvee commented on install_config/registry/index.adoc:
```diff
@@ -45,3 +45,112 @@ If you need an enterprise-quality container image registry, Red Hat Quay is avai
+. Copy the token.
+
+[[using-service-accounts]]
+=== Using Service Accounts
```
Specify these are the manual steps

bmcelvee commented on install_config/registry/index.adoc:
```diff
@@ -45,3 +45,112 @@ If you need an enterprise-quality container image registry, Red Hat Quay is avai
+.. Submit the secret to the cluster, for example:
++
+----
+$ kubectl create -f <account-name>-secret.yml --namespace=NAMESPACEHERE
```
oc create

bmcelvee replied:
s/--namespace=NAMESPACEHERE/--namespace=openshift

bmcelvee commented on install_config/registry/index.adoc:
```diff
@@ -45,3 +45,112 @@ If you need an enterprise-quality container image registry, Red Hat Quay is avai
 Visit the link:https://quay.io[Quay.io] site to set up your own hosted Quay registry account. After that, the link:https://quay.io/tutorial/[Quay Tutorial] helps you login to the Quay registry and start managing your images. Alternatively, refer to link:https://access.redhat.com/documentation/en-us/red_hat_quay/2.9/html-single/getting_started_with_red_hat_quay/[Getting Started with Red Hat Quay]  ...
 
 At the moment, you access your Red Hat Quay registry from OpenShift as you would any remote container image registry. To learn how to set up credentials to access Red Hat Quay as a secured registry, refer to xref:../../dev_guide/managing_images.adoc#allowing-pods-to-reference-images-from-other-secured-registries[Allowing Pods to Reference Images from Other Secured Registries].
+
```
Move up one level and specify internal registry versus external

bmcelvee commented on install_config/registry/index.adoc:
```diff
@@ -37,11 +37,3 @@ installed separately as a stand-alone container image registry.
 To install a stand-alone registry, follow
 xref:../../install/stand_alone_registry.adoc#install-config-installing-stand-alone-registry[Installing a Stand-alone Registry]. This installation path deploys an all-in-one cluster
 running a registry and specialized web console.
-
```
@chrisnegus Since this topic is for the internal registry, I moved the TBR (auth enabled) and Quay content out. I moved the Quay content to the Architecture guide:  architecture/infrastructure_components/image_registry.adoc

bmcelvee commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,94 @@
+`openshift_additional_registry_credentials` variable.
+* Alerts you if you did not provide credentials and are using `registry.redhat.io`
+for either `openshift_examples_registryurl` or `oreg_url`, since it defaults to
+`registry.redhat.io`.
```
Thank you!

bmcelvee commented:
@wenjing PTAL

bmcelvee commented:
New preview builds 
Architecture: http://file.rdu.redhat.com/~bmcelvee/21092018/terms-based-registry/architecture/infrastructure_components/image_registry.html

Accessing and Configuring the Red Hat Registry: http://file.rdu.redhat.com/~bmcelvee/21092018/terms-based-registry/install_config/configuring_red_hat_registry.html

vikram-redhat commented:
@bmcelvee we need to send this to QE ASAP.

bmcelvee commented:
@vikram-redhat, yes. I requested review from @wenjing last week. 

Thanks!

bmcelvee commented:
@openshift/team-documentation PTAL

kalexand-rh commented on architecture/infrastructure_components/image_registry.adoc:
```diff
@@ -46,11 +45,24 @@ creation.  Refreshing the fetched tags is as simple as running `oc import-image
+[[red-hat-quay]]
+== Red Hat Quay Registries
+
+If you need an enterprise-quality container image registry, Red Hat Quay is available both as a hosted service and as software you can install in your own data center or cloud environment. Advanced registry features in Red Hat Quay include geo-replication, image scanning, and the ability to rollback images.
```
s/rollback/roll back

kalexand-rh commented on architecture/infrastructure_components/image_registry.adoc:
```diff
@@ -46,11 +45,24 @@ creation.  Refreshing the fetched tags is as simple as running `oc import-image
+
+If you need an enterprise-quality container image registry, Red Hat Quay is available both as a hosted service and as software you can install in your own data center or cloud environment. Advanced registry features in Red Hat Quay include geo-replication, image scanning, and the ability to rollback images.
+
+Visit the link:https://quay.io[Quay.io] site to set up your own hosted Quay registry account. After that, the link:https://quay.io/tutorial/[Quay Tutorial] helps you login to the Quay registry and start managing your images. Alternatively, refer to link:https://access.redhat.com/documentation/en-us/red_hat_quay/2.9/html-single/getting_started_with_red_hat_quay/[Getting Started with Red Hat Quay]  ...
```
I'd s/the link:https://quay.io/tutorial/[Quay Tutorial] helps you login/follow the link:https://quay.io/tutorial/[Quay Tutorial] to log in
extra space before refer
s/on setting/about setting

kalexand-rh commented on architecture/infrastructure_components/image_registry.adoc:
```diff
@@ -46,11 +45,24 @@ creation.  Refreshing the fetched tags is as simple as running `oc import-image
+
+Visit the link:https://quay.io[Quay.io] site to set up your own hosted Quay registry account. After that, the link:https://quay.io/tutorial/[Quay Tutorial] helps you login to the Quay registry and start managing your images. Alternatively, refer to link:https://access.redhat.com/documentation/en-us/red_hat_quay/2.9/html-single/getting_started_with_red_hat_quay/[Getting Started with Red Hat Quay]  ...
 
+At the moment, you access your Red Hat Quay registry from OpenShift as you would any remote container image registry. To learn how to set up credentials to access Red Hat Quay as a secured registry, refer to xref:../../dev_guide/managing_images.adoc#allowing-pods-to-reference-images-from-other-secured-registries[Allowing Pods to Reference Images from Other Secured Registries].
```
s/At the moment, you access your Red Hat Quay registry from OpenShift as you would any/You can access your Red Hat Quay registry from {product-title} like any

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+
+//Concept
+All container images available through the Red Hat Container Catalog are hosted
+on an image registry, `registry.access.redhat.com`. The Red Hat Container Catalog
```
There are several double spaces in this PR. Please check.

I'd say "With {product-title} version 3.11, the Red Hat Container Catalog moved from..."

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+is moving from `registry.access.redhat.com` to `registry.redhat.io`.
+
+The new registry, `registry.redhat.io`, requires authentication for access to
+images and hosted content on {product-title}. Following the move to the new
```
"Following the move to the new registry, the existing registry will be available for a period of time."

Can we be more specific? If not, I'd remove it.

bmcelvee replied:
This is a tough one. I left it intentionally vague. We need to let readers know that the existing registry will eventually be removed. I could maybe rewrite it. 

The goal is 12 months. But if we make that statement and it changes, we will be responsible for maintaining it and could run into further problems.

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+
+[NOTE]
+====
+{product-title} pulls images from `registry.redhat.io`, so you will need to
```
... so you must configure your cluster to use it.

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+The new registry uses standard OAuth mechanisms for authentication,
+with the following methods:
+
+* *Web username and password.* This is the standard set of credentials you use
```
I'd make these dls.

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+accounts that give systems the ability to authenticate against
+the container registry.
+
+You can use `docker login` with your credentials, username and password or authentication
```
s/username/either user name

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+You can use `docker login` with your credentials, username and password or authentication
+token, to access content on the new registry.
+
+All image streams are updated to point to the new registry. Because the
```
s/are updated to/

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+
+All image streams are updated to point to the new registry. Because the
+new registry requires authentication for access, there is a new secret in the
+OpenShift namespace called `imagestreamsecret`.
```
Please confirm that all the "OpenShift" references in this PR don't need to be "{product-title}"

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+new registry requires authentication for access, there is a new secret in the
+OpenShift namespace called `imagestreamsecret`.
+
+There are two places your credentials must exist:
```
You must place your credentials in two places:

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+There are two places your credentials must exist:
+
+* *OpenShift namespace*. Your credentials must exist in the OpenShift
+namespace so that the image streams in the OpenShift namespace can import.
```
maybe s/so that the image streams in the OpenShift namespace can import./so that the images streams that the namepace references can import

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+
+* *OpenShift namespace*. Your credentials must exist in the OpenShift
+namespace so that the image streams in the OpenShift namespace can import.
+* *Your host*. Your credentials must exist on your host because Kubernetes will
```
s/will use/uses

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+* *Your host*. Your credentials must exist on your host because Kubernetes will
+use the credentials from your host when it goes to pull images.
+
+As long as you have the following items in place, you should be able to
```
To access the new registry:

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+As long as you have the following items in place, you should be able to
+access the new registry.
+
+* You should see image import secret, `imagestreamsecret`,  in your OpenShift
```
Please check for modals and unnecessary future tense.

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+* All of your cluster nodes should have a `dockerconfigjson` that allows you to
+access the Red Hat registry.
+
+You can also manage these credentials during installation or upgrade using the
```
I don't think 66-75 belong in a concept.

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+value of `registry.redhat.io` for either `openshift_examples_registryurl` or `oreg_url`.
+endif::[]
+
+Clusters that require access to additional authenticated registries can configure
```
This paragraph is also blurring concept and procedure.

kalexand-rh commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,114 @@
+openshift_additional_registry_credentials=[{'host':'registry.example.com','user':'name','password':'pass1','test_login':'False'},{'host':'registry2.example.com','password':'token12345','tls_verify':'False','image_name':'mongodb/mongodb'}]
+----
+
+[[create-user-accounts]]
```
I'd make this part of the procedure, not the concept.

kalexand-rh commented on architecture/infrastructure_components/image_registry.adoc:
```diff
@@ -46,11 +45,24 @@ creation.  Refreshing the fetched tags is as simple as running `oc import-image
+[[auth-enabled-registry]]
+== Authentication Enabled Red Hat Registry
+
+include::architecture/topics/auth_enabled_registry.adoc[leveloffset=+1]
```
The heading should be in the module, and this should be leveloffset=+2

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+include::architecture/topics/auth_enabled_registry.adoc[leveloffset=+1]
+
+
+[[create-authentication-tokens]]
```
This section should be a module.

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+organization.
+
+//[[Prerequisites]]
+*Prerequisites*
```
.Prerequisites and .Procedure

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+//[[Prerequisites]]
+*Prerequisites*
+
+* User credentials
```
yeah, getting creds should be a module and be called before this module

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+To create a token in order complete a `docker login`:
+
+. Navigate to `registry.redhat.io`.
+. Log in with your RHN username and password.
```
Expand the abbreviation on first use.

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+. Navigate to `registry.redhat.io`.
+. Log in with your RHN username and password.
+. Accept terms when prompted.
+.. If you are not immediately prompted to accept terms, you will be prompted
```
That's a note, not a step.

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+[[using-service-accounts]]
+=== Using Service Accounts
+
+Once you have created your service accounts and generated tokens, you can perform
```
s/Once/After

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+=== Using Service Accounts
+
+Once you have created your service accounts and generated tokens, you can perform
+additional tasks.
```
What additional tasks?

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+[NOTE]
+====
+This section provides the manual steps, which can be automatically performed
+during installation.
```
Does this mean that if you provide the relevant inventory parameters and values that you don't have to do this? I'd be a little more explicit.

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+following tasks:
+
+* Under the *Token Information* tab, you can view your username (the name you provided
+ prepended with a random string) and password (token). From this tab, you can
```
Why would I need to regenerate a token?

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+* Under the *Token Information* tab, you can view your username (the name you provided
+ prepended with a random string) and password (token). From this tab, you can
+ regenerate your token.
+* Under the *OpenShift Secret* tab, you can:
```
s/Under/From

Do I need to do this? If so, s/, you can:/:

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+ regenerate your token.
+* Under the *OpenShift Secret* tab, you can:
+.. Download the secret by clicking the link in the tab.
+.. Submit the secret to the cluster, for example:
```
s/, for example:/:

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+    imagePullSecrets:
+      - name: <numerical-string-account-name>-pull-secret
+----
++
```
Can we break these out into separate steps instead of nesting actions under the example?

kalexand-rh commented on install_config/configuring_red_hat_registry.adoc:
```diff
@@ -0,0 +1,127 @@
+  systemctl restart atomic-openshift-node
+----
++
+* Under the *Docker Configuration* tab, you can:
```
And then what?

bparees commented on install_config/topics/creating_service_accounts.adoc:
```diff
@@ -0,0 +1,43 @@
+// * Title: = Doing procedure A
+
+[id='creating-service-accounts-authentication-tokens_{configuring_red_hat_registry}']
+= Creating Service Accounts and Authentication Tokens
```
we need to be really really careful to disambiguate between "service accounts/tokens for the redhat registry" and "openshift cluster service accounts/tokens".

they are entirely unrelated.

bparees commented on install_config/topics/using_service_accounts.adoc:
```diff
@@ -0,0 +1,94 @@
+// * Title: = Doing procedure A
+
+[id='using-service-accounts_{configuring_red_hat_registry}']
+= Using Service Accounts
```
same comment here.

dongboyan77 commented on architecture/infrastructure_components/image_registry.adoc:
```diff
@@ -36,21 +36,42 @@ container registry, without the build and deployment integration. See
+You can access your Red Hat Quay registry from {product-title} like any remote
+container image registry. To learn how to set up credentials to access
+Red Hat Quay as a secured registry, refer to xref:../../dev_guide/managing_images.adoc#allowing-pods-to-reference-images-from-other-secured-registries[Allowing Pods to Reference Images from Other Secured Registries].
+
```
L48-65 look good to me

wzheng1 commented on architecture/topics/auth_enabled_registry.adoc:
```diff
@@ -0,0 +1,64 @@
+* Verify image import secret, `imagestreamsecret`, is in your OpenShift
+namespace. That secret has credentials that allow you to access
+the new registry.
+* Verify all of your cluster nodes have a `dockerconfigjson` that allows you to
```
Could we update  add an explanation to "dockerconfigjson" like the file is copy from "~/.docker/config.json" on master.

jianlinliu replied:
To be more precise, how about change `dockerconfigjson` to `/var/lib/origin/.docker/config.json`

jianlinliu commented on install_config/topics/using_service_accounts.adoc:
```diff
@@ -0,0 +1,94 @@
+`/var/lib/origin/.docker/config.json` and restart the node.
++
+----
+# cp ~/.docker /var/lib/origin/
```
s/cp/cp -r/

jianlinliu commented on install_config/topics/managing_registry_credentials_install.adoc:
```diff
@@ -0,0 +1,72 @@
+For example:
+
+----
+openshift_additional_registry_credentials=[{'host':'registry.example.com','user':'name','password':'pass1','test_login':'False'},{'host':'registry2.example.com','password':'token12345','tls_verify':'False','image_name':'mongodb/mongodb'}]
```
s/image_name/test_image/

bmcelvee commented:
Since this PR has been through so many reviews, I am going to go ahead and merge and CP it with the to_followup label. If there are any additional edits that need to be made to this content, I can open a follow-up PR. 

Thanks!

bmcelvee commented:
/cherrypick enterprise-3.11
